File: ct_206/Project 3/project3.py
# Create dictionary with 5 values
dictionary = {
  "a": 1,
  "b": 2,
  "c": 3,
  "d": 4,
  "e": 5
}
print(dictionary)

# Add new key value pair
dictionary["f"] = 6
print(dictionary)

# Change value of key a
dictionary["a"] = 0
print(dictionary)

# Print 3 values via their key
print("Accessing via key:", dictionary["a"], dictionary["b"], dictionary["c"])

# Print 3 keys via their value
print("Accessing via value:", \
      list(dictionary.keys())[list(dictionary.values()).index(0)], \
      list(dictionary.keys())[list(dictionary.values()).index(2)], \
      list(dictionary.keys())[list(dictionary.values()).index(4)])

# Delete an element from the dictionary
del dictionary["a"]
print(dictionary)

# Test for the inclusion of 3 elements
if dictionary["b"] and dictionary["c"] and dictionary["d"]:
  print("True")
else:
  print("False")

# Printing all keys and values in the dictionary
print("Looping through values:")
for key, value in dictionary.items():
  print(key, ":", value)

File: ct_206/Project 3/project3.pl
# Create a hash with 5 values
my %hash = (
  "a" => 1,
  "b" => 2,
  "c" => 3,
  "d" => 4,
  "e" => 5
);

print "@{[%hash]}\n";

# Add a new key/value pair to the dictionary
$hash{"f"} = 6;

print "@{[%hash]}\n";

#  Change on of the values through reassignment
$hash{"a"} = 0;

print "@{[%hash]}\n";

# Access at least 3 values using the keys
print "Accessing via key: $hash{\"a\"} $hash{\"b\"} $hash{\"c\"}\n";

# Access at least 3 keys using the values
my %rhash = reverse %hash;
print "Accessing via value: $rhash{0} $rhash{2} $rhash{3}\n";

# Delete an element
delete $hash{"a"};

print "@{[%hash]}\n";

# Test for key inclusion for 3 elements
if (exists $hash{"b"} && exists $hash{"c"} && exists $hash{"d"}) {
  print "True\n";
} else {
  print "False\n";
}

# Loop through the dictionary, printing out
# the key and value on the same line separated
# by a space.
while ((my $key, $value) = each %hash) {
  print "$key, $value\n";
}